[Feat] #17 - saved detail 구현

ghi512 opened the pull request:
### Pull request
⛳️ 작업한 브랜치
- <#17>

👷 작업한 내용
- Saved 페이지 구현
- CafeDetail 페이지 구현
- ReviewWrite 페이지 구현
- 홈 카페 팝업 - 상세 페이지 연결

🔥 변경 사항
1. 컴포넌트 추가 및 수정
- CafeDetail, ReviewWrite, Saved, SavedCafeCard, DetailReviewCard, ImageCarousel 추가
- Bookmark: width, height 받아 설정하도록 수정
- TopBar: TagGroup만 있을 때의 Top 높이 조절
- CafeInformationDetail: CSS 수정 (너비, 정렬, 업뎃시간 위치)

2. 라우팅 업데이트
- /cafe/:id: 카페 상세 페이지
- /cafe/:id/review-write: 리뷰 작성 페이지

3. 이미지 슬라이더 기능 구현을 위한 react-slick 라이브러리 설치

### 스크린샷
|기능|스크린샷|
|---|------|
|Saved|<img src="https://github.com/user-attachments/assets/d8cf69c7-2ddb-476d-aace-5f5a890b2de5" width="250px">|
|CafeDetail|<img src="https://github.com/user-attachments/assets/710d91c3-221b-4aae-ae38-e0ac8f9adf3a" width="250px">|
|ReviewWrite|<img src="https://github.com/user-attachments/assets/133394e1-8db6-4329-b91e-5fae42e96dae" width="250px">|
|홈 카페 팝업 - 상세 페이지 연결|<img src="https://github.com/user-attachments/assets/bc7b9c78-5e63-4238-a808-ebb117f7ffed" width="250px">|

### comment
- CafeDetail 페이지에서 처음 상세정보 탭을 클릭했을 때에는 지도에 잘 표시되지만, 리뷰 탭을 선택했다 돌아오면 지도가 표시되지 않는 문제가 있습니다. 원인 파악 중입니다 ... 😭

### 관련 이슈
- <#17>

ghi512 committed:
[fix] #17 - TopBar - tag-select-group top 동적으로 설정

ghi512 committed:
[feat] #17 - SavedCafeCard 구현

ghi512 committed:
[feat] #17 - Saved 페이지 구현

ghi512 committed:
[feat] #17 - saved 북마크 버튼 추가, 카페 카드 눌렀을 때 detail로 이동하도록

ghi512 committed:
[feat] #17 - index.js 컴포넌트 추가

ghi512 committed:
[fix] #17 - index.js 중복 정리

ghi512 committed:
[fix] #17 - CafeInformationDetail 너비, 정렬, 업뎃시간 위치 조절

ghi512 committed:
[feat] #17 - SearchResult 클릭 효과 여부 설정 추가

ghi512 committed:
[feat] #17 - ImageCarousel 추가

이미지 슬라이더 기능 구현을 위함 (react-slick 라이브러리 설치)

ghi512 committed:
[feat] #17 - CafeDetail 구현

ghi512 committed:
[add] #17 - detail 정보 아이콘 추가

ghi512 committed:
[feat] #17 - CafeDetail info탭 구현

ghi512 committed:
[fix] #17 - TopBar 사진 경로를 루트(/)로 변경

ghi512 committed:
[feat] #17 - ReviewWrite 구현

ghi512 committed:
[feat] #17 - CafeDetail 뒤로가기 버튼 추가

ghi512 committed:
[feat] #17 - DetailReviewCard 구현

ghi512 committed:
[feat] #17 - CafeDetail 리뷰 탭 구현

ghi512 committed:
[feat] #17 - Home 카페 팝업 상세보기 연결

jjinleee commented:
수고하셨습니당🙌🏻

### Review

1. Saved 탭에서 카페 img크기를 줄이고 간격을 넓히면 좋을 것 같아요

|현재|피그마|
|---|------|
|![image](https://github.com/user-attachments/assets/f3228c1a-d101-4515-82e8-4f7d0a518834)|![image](https://github.com/user-attachments/assets/806060ac-894d-481d-a32b-3b0f05bc241b)|

2. 피그마에선 적용되어 있지 않지만 saved에 있는 카페들은 이미 저장된 카페이므로 bookmark는 모두 채워져 있어야 합니다. 카페 상세페이지도요!
3. 카페 상세 페이지에서 cafe-detail-name, cafe-detail-address의 폰트 크기와 색상이 다릅니다. 피그마와 같게 수정부탁드려요! 또한 아래사진처럼 상하 간격이 같으면 좋을 것 같아요
![image](https://github.com/user-attachments/assets/f8a03df1-edc9-4fa9-a86e-c602ebebcc69)

4. 상세정보와 리뷰 탭 왼쪽에 margin 추가하는게 좋을 것 같습니다. 저도 margin 추가했어요!(margin-left: 10px;). 또한 이부분도 폰트 크기가 달라요
 ![image](https://github.com/user-attachments/assets/9fd005de-fd3d-47ea-902d-58c11c65e73b)

5.  리뷰쓰기 버튼의 크기도 피그마와 같게 수정해주시고 폰트크기도 피그마와 달라 수정이 필요합니다!!



+) 폰트 크기와 bold 는 피그마와 저희가 정한 font weight를 참고해서 같게 해주시면 좋을 것 같아요. 변경하고 싶으시다면 먼저 말씀해주세용
+) 버튼 크기나 패딩 height, weight등도 피그마와 비슷하게 해주시면 감사하겠습니다🫶🏻

ghi512 committed:
[fix] #17 - Saved 카페 사진 간격 조정

ghi512 committed:
[fix] #17 - CafeDetail 타이틀 css 수정

cafe-detail-name, cafe-detail-address 폰트 색상, 크기, font-weight 수정 / 이미지와 구분선 간의 간격 조정

ghi512 committed:
[feat] #17 - cafe-detail-tab css 수정

margin-left 추가, 폰트크기 수정

ghi512 committed:
[fix] #17 - 리뷰 버튼 크기, 폰트 사이즈 수정

ghi512 commented:
@jjinleee 

1. Saved 탭 이미지 간 간격 넓혔습니다 !
<img width="302" alt="image" src="https://github.com/user-attachments/assets/f013dab7-47a9-4184-93c5-a7bde4418ba4" />

2. 상세 정보페이지 - 카페명, 주소의 폰트 색상, 크기, 간격 / 탭 왼쪽 마진, 폰트 크기 / 리뷰쓰기버튼 크기, 폰트 사이즈 수정 완료했습니다!
<img width="380" alt="image" src="https://github.com/user-attachments/assets/c962a4b2-befc-4c4e-9032-57a5a7cdc0d6" />
<img width="390" alt="image" src="https://github.com/user-attachments/assets/5ee01bf3-592c-418f-8c58-0a2fc6452267" />

감사해용~~💕

jjinleee commented:
|현재|피그마|
|--|---|
|<img width="286" alt="image" src="https://github.com/user-attachments/assets/060004b2-6b6e-472c-a593-48d66394c035" /> |<img width="232" alt="image" src="https://github.com/user-attachments/assets/a24eb537-8e8c-47e2-8f94-bce2b875f376" />|

이페이지 전체적으로 피그마와 비슷하게 패딩 추가부탁드려용🫶🏻

ghi512 committed:
[fix] #17 - 카페 데이터 전역 관리 및 Context API 적용

ghi512 committed:
[fix] #17 - 저장된 카페 CafeContext와 연동 및 북마크 채운 상태로 표시

ghi512 committed:
[fix] #17 - CafeDetail 탭 전환 시 지도 초기화 문제 해결

ghi512 committed:
[fix] #17 - CafeDetail css 수정

ghi512 commented:
@jjinleee 아래 내용 수정했습니다 ~~

**🎯 1. home 검색창 - 카카오맵 api내 카페가 아닌 TagCafe 자체 데이터로 카페 검색되게 수정**
- react에서 제공하는 Context를 사용(```CafeContext.js```)하여 전역 데이터 사용 (추후 DB 데이터로 수정 필요)
- Home.js, App.js, Search.js, CafeDetail.js, CafePopup.js 파일 수정
<br>

**🎯 2. 저장된 카페 CafeContext와 연동 및 북마크 채운 상태로 표시되도록 수정**
- Saved에 하드코딩되었던 카페 목록 대신, CafeContext에서 saved가 true인 경우 Saved에 카페가 표시되도록 수정
- SavedCafeCard.js에서 북마크가 항상 저장된 상태로 보이도록 수정
- 저장된 카페 목록(Saved), 상세 페이지(CafeDetail)에서 북마크 해제하면 saved:false로 변경되고 목록에서 제거됨
- 저장된 카페가 1개일 때 카드가 너무 넓어지는 문제 해결 (auto-fill 적용) 
![Jan-30-2025 16-50-22](https://github.com/user-attachments/assets/9ebe0ba0-cd00-4a29-b434-64c6421cef54)
<br>

**🎯 3. CafeDetail 탭 전환 시 지도 초기화 문제 해결**
- 탭 전환 시 지도가 뜨지 않던 문제 해결했습니당
![Jan-30-2025 16-46-48](https://github.com/user-attachments/assets/49dacacd-16b9-4786-968f-6decdee62eb9)
<br>

**🎯 4. CafeDetail(카페 상세페이지) css 수정**
<img width="292" alt="image" src="https://github.com/user-attachments/assets/fa747081-45fa-4546-9730-6d2b2d1447b3" />
<br>

더 수정 필요한 부분 있으면 말씀해주세요 ! 감사합니다 🫶

jjinleee commented:
수고하셨습니다~

## REVIEW
1. 카페를 검색후 상세페이지 들어가서 리뷰 작성을 하면 카페이름, 주소가 뜨지 않습니다. 또한 리뷰작성 페이지에서 뒤로가기를 누르면 에러가 발생합니다. 아마 카페 이름을 읽는 부분에서 에러가 발생한 것 같습니다

|카페 상세 내용(이름, 주소) 없음|에러내용|
|---|------|
|![comment](https://github.com/user-attachments/assets/c32248a0-40c5-490c-b7bc-78497f613d2e)|<img width="635" alt="image" src="https://github.com/user-attachments/assets/1a335dda-5675-4a58-a81e-dd1a9b5bde63" />|

2. 저장한 카페들은 새로고침 후 초기화되는데 이건 DB로 교체하면 괜찮겠죠?

3. Home에서 전체 필터를 한번에 클릭했을때 개별 태그에 내용이 해당 내용이 들어가야 할 것 같아요. 이건 건의사항입니다!
<img width="302" alt="image" src="https://github.com/user-attachments/assets/09203383-41c2-47a6-b1d9-f0b9bf77928c" />

이것 외에는 문제 없어보이니 먼저  승인해놓겠습니다.
해당 에러 픽스후에 머지부탁드려요👏🏻👏🏻

ghi512 committed:
[fix] #17 - ReviewWrite 카페 전역 상태 적용 & 뒤로가기 오류 수정

ghi512 committed:
[feat] #17 - 개별 태그와 TagFilter 상태 동기화

ghi512 commented:
1. 제가 리뷰작성 부분 수정하는 걸 깜빡했네요 .. 😵‍💫 CafeContext 사용하도록 수정했습니다 !
![리뷰작성 수정](https://github.com/user-attachments/assets/b119ed01-af5f-4523-ac8d-754c68248643)

2. 네네! 그 부분은 DB 연동해야 해용

3. 수정했습니다 ~~
![태그 수정](https://github.com/user-attachments/assets/bebe28b2-bf6d-43e7-b62b-902968ca6de9)

다시 승인 부탁드려요 히히 🥸 @jjinleee
